### ISSUE TITLE
What's the correct way to tear down a kogrid instance?

### ISSUE THREAD

**Issue by thcgit** [Knockout-Contrib/KoGrid#122]
Hi there,

I have a single page app that utilizes kogrid. When I navigate to another view, I want to tear down (clean up) the kogrid instance.

I tried using ko.cleanNode but I found that a lot of memory was still being used. Removing all objects in the grid's data source array helps but there are still kogrid objects remaining in memory.

See http://jsfiddle.net/qH5M3/2/

After adding and removing the grid, an instance of my view model and several kg objects remain in the object heap if you do a profile.

**Comment by orneryd** [comment]
We need to add documentation but there is a window namespace service call kg.gridService

you can call     RemoveGrid: function(gridId) {
        delete kg.gridService.gridCache[gridId];
    },

you can get the gridId from the grid.

However, this shouldn't be a problem in kogrid 2.0

**Comment by thcgit** [comment]
Hi Tim,

I spent some time testing KoGrid 2.0 and it looks like when I ko.clean my element and call kg.gridService.RemoveGrid, the grid and my view model is still not being garbage collected.

I made some changes locally and that has fixed the problem for me. I see that you're still fixing up merging and other issues so I'll post what I did here instead of making a pull request.

When removing the grid in RemoveGrid:
1. Remove all ko subscriptions on properties on the grid.
2. Dispose all computed observables on the grid.
3. Clear grid.columns.
4. Unhook the window.resize event. I changed the event to a namespaced one so that I could unhook just the one that was created for my grid.

Doing all of the above removes all references to the grid and allows it to be collected properly.

Also, an unrelated issue is that in kg.StyleProvider, grid.topPanelStyle is being set twice so the first computed that is being created is redundant.
